### DISCUSSION TITLE
Memory bloat issue observed while using cursor with execute while fetching table data.

### DISCUSSION THREAD

**Discussion by neel004** [sqlalchemy/sqlalchemy#11077] [Usage Questions] [answered]
I used `execute` method from Sqlalchemy for Postgres with asyncpg lib.  I'm observing a severe spike in memory usage depending on the size and number of total records present in the table when I try to query the table with the cursor.fetchmany method.
I think that upon querying the table data, even when we use the cursor object it scrapes whole table data at once in memory and then iterates over it from the memory itself even though it should perform a network call for each page.

**Comment by CaselIT** [comment, marked as answer]
Hi,

What you are observing is documented behaviour. Please take a look at the documentation https://docs.sqlalchemy.org/en/20/orm/extensions/asyncio.html#synopsis-core in particular stream

**Comment by neel004** [comment]
So, To have a streaming response, i.e. new network call for each page, are you suggesting to use the [stream()](https://docs.sqlalchemy.org/en/20/orm/extensions/asyncio.html#sqlalchemy.ext.asyncio.AsyncConnection.stream) instead of execute() with fetchmany()? in that case, I can see stream() also has the fetchmany() method available would it make a new API call for each page or it would behave similar to execute()

**Reply by CaselIT** [reply]
I've not 100% understood why you mean, but execute does pre-buffer every row

**Reply by neel004** [reply]
Thanks for the quick replies. 🙌🏻 made it to work with a work around.
